src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -408,12 +466,36 @@ namespace winrt::TerminalApp::implementation
     void CommandPalette::_filterTextChanged(IInspectable const& /*sender*/,
                                             Windows::UI::Xaml::RoutedEventArgs const& /*args*/)
     {
+        if (_currentMode != CommandPaletteMode::TabSwitcherMode)
+        {
+            _checkActionVsCommandlineMode();
+        }
+
         _updateFilteredActions();
         _filteredActionsView().SelectedIndex(0);
 
         _noMatchesText().Visibility(_filteredActions.Size() > 0 ? Visibility::Collapsed : Visibility::Visible);
     }
 
+    void CommandPalette::_checkActionVsCommandlineMode()
```

Comment:
nit: could we rename this to "_evaluatePrefix" or something more extensible? As we add more modes, I wouldn't want to rename this "_checkActionVsCommandlineVsProfileSelectorVsTabSwitcherMode" haha

Reply:
Sure why not, I hated that method name anyways